src/components/users/profile/ko/runtime/profile.ts

```diff
@@ -128,31 +119,13 @@ export class Profile {
             return;
         }
         
-        this.cleanValidationErrors();
         this.working(true);
 
-        try {
+        await tryCatchDispatchError(async () => {
```

Comment:
`await tryCatchDispatchError(async () => {` - I can't say I like this pattern. This maybe Ok for these cases, but if we want to handle other errors, not related to validation, this becomes problematic.

Maybe we could do something like this (keeping your new utils):

```
try {
   ...
}
catch (error) {
    if (error instanceof ValidationError) {
         handleValidationError(this.eventManager, errorSources.signInOAuth, error);
    }
    else if (error instanceof OtherError) {
       ...
    }
}
finally {
   ...
}
```

(we don't have ValidationError class, but MapiClient could throw one when the code is "ValidationError" instead of generic MapiError)

Reply:
I wasn't sure about this part either, why I included it in the end was only for several repetitive cases of 
```
dispatchErrors(eventManager, source, []); // clear error
try {
    foo();
  } catch (error) {
    // dispatch only the error and don't do anything else
    return parseAndDispatchError(eventManager, source, error, fallbackMessage, errorDetailsMap);
}
```
but yes, if requirement change and we need to do something in catch block, it might be confusing what to do so I removed the function so it's more verbose but clearer

Reply:
You can see nice example of what I originally had in mind in this commit: https://github.com/Azure/api-management-developer-portal/pull/1799/commits/748afedc34201942a7cc29d7e3c05f77abd69099